doltpy/core/dolt.py

```diff
@@ -72,144 +65,215 @@ def __str__(self):
+    def __init__(self, repo_dir: str):
         self._repo_dir = repo_dir
         self.server = None
 
-    def repo_dir(self) -> str:
+    def repo_dir(self):
         return self._repo_dir
 
-    def config(self, is_global: bool, user_name: str, user_email: str):
-        """
-        Exposes a way to set the user name and email to be associated with commit messages. Can be either global, or
-        local to this repo.
-        :param is_global:
-        :param user_name:
-        :param user_email:
-        :return:
-        """
-        args = ["dolt", "config", "add"]
-        if is_global:
-            args.append("--global")
-
-        name_args = args
-        email_args = args.copy()
+    def execute(self, args: List[str], print_output: bool = True) -> List[str]:
+        output = _execute(args, self.repo_dir())
 
-        name_args.extend(["user.name", user_name])
-        email_args.extend(["user.email", user_email])
+        if print_output:# TODO configure outpu
+            print(output)
 
-        if is_global:
-            _execute(args=name_args, cwd=None)
-            _execute(args=email_args, cwd=None)
-
-        else:
-            _execute(args=name_args, cwd=self.repo_dir())
-            _execute(args=email_args, cwd=self.repo_dir())
+        return output.split('\n')
 
-    def create_branch(self, branch_name: str, commit_ref: str = None):
+    @staticmethod
+    def init(repo_dir: str = None) -> 'Dolt':
         """
-        Creates a new branch in this repo with the name branch_name. If commit_ref is None the ref is the HEAD of the
-        currently checked out branch.
-        :param branch_name:
-        :param commit_ref:
+        Creates a new repository in the directory specified, creating the directory if `create_dir` is passed, and returns
+        a `Dolt` object representing the newly created repo.
         :return:
         """
-        args = ["dolt", "branch", branch_name]
+        if not repo_dir:
+            repo_dir = os.getcwd()
 
-        if commit_ref is not None:
-            args.append(commit_ref)
+        if os.path.exists(repo_dir):
+            logger.info('Dolt repo in existing dir {}'.format(repo_dir))
+        else:
+            try:
+                logger.info('Creating directory {}'.format(repo_dir))
+            except Exception as e:
+                raise e
 
-        _execute(args=args, cwd=self.repo_dir())
+        output = _execute(['init'], cwd=repo_dir)
+        print(output)
+        return Dolt(repo_dir)
 
-    def checkout(self, branch_name: str):
-        """
-        Check out the repo in `self.repo_dir()` at the specified branch.
-        :param branch_name: the branch to checkout at
-        :return:
-        """
-        assert branch_name in self.get_branch_list(), 'Cannot checkout of non-existent branch {}'.format(branch_name)
-        args = ["dolt", "checkout", branch_name]
-        _execute_restart_serve_if_needed(self, args)
+    def status(self) -> DoltStatus:
```

Comment:
I think we could do better here by adding a `-v`/`--verbose` option to `dolt status` that gives us more specific information. 
Perhaps we could combine `ls()` and `status()` into some sort of function or object that tracks the whole repo state.

Reply:
That's a good shout, elsewhere you might have seen that we use verbose by default and then parse the information into helpful objects, I think we should do the same here!